flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/JoinITCase.scala

```diff
@@ -372,11 +366,219 @@ class JoinITCase(
 
     val sqlQuery1 = "SELECT * FROM A CROSS JOIN (SELECT count(*) FROM A HAVING count(*) < 0)"
     val result = tEnv.sql(sqlQuery1).count()
-
     Assert.assertEquals(0, result)
   }
 
   @Test
+  def testLeftNullLeftJoin (): Unit = {
```

Comment:
We need to check the following cases:
- LEFT JOIN, single row on the right
- LEFT JOIN, empty input on the right
- RIGHT JOIN, single row on the left
- RIGHT JOIN; empty input on the left